### PR TITLE
feat(xchain): adds read and write network and requestKey from querystring params

### PR DESCRIPTION
**Pull request by alber70g**
Add querystring params for `network` and `requestkey`

### CODE REVIEW

File: docs/xchain.js [hunk 1]
```diff
@@ -1033,3 +1040,36 @@ function makeRawRequestInit(stringBody) {
     body: stringBody,
   };
 }
+
+function fillNetworkIdFromQueryString() {
+  const requestKey = getParameterByName('network');
+  if (requestKey) {
+    document.getElementById('pact-id').value = requestKey;
```
Comment (sanderlooijenga) [inline docs/xchain.js:1047]: I think this should be `document.getElementById('networkId')`

File: docs/xchain.js [hunk 2]
```diff
@@ -1033,3 +1040,36 @@ function makeRawRequestInit(stringBody) {
...
+  }
+}
+
+function getParameterByName(name, url = window.location.href) {
+  name = name.replace(/[\[\]]/g, '\\$&');
+  const regex = new RegExp(`[?&]${name}(=([^&#]*)|&|#|$)`);
+  const results = regex.exec(url);
+  if (!results) return null;
```
Comment (sanderlooijenga) [inline docs/xchain.js:1055]: Should we also return an empty string here?

File: docs/xchain.js (hunk 1 above)
Comment (alber70g) [reply docs/xchain.js:1055]: This is the `requestKey`. The element for that has id `pact-id`

File: docs/xchain.js (hunk 2 above)
Comment (alber70g) [reply docs/xchain.js:1064]: Copilot told me so :')

File: docs/xchain.js (hunk 1 above)
Comment (sanderlooijenga) [reply docs/xchain.js:1055]: I think the `const` is named wrong, since you're retrieving the `network` parameter 😁

File: docs/xchain.js (hunk 2 above)
Comment (sanderlooijenga) [reply docs/xchain.js:1064]: Alrighty 👌

**Review by sanderlooijenga — APPROVED** [review]

**Pull request merged**